crates/bevy_ecs/src/system/system_registry.rs

```diff
@@ -0,0 +1,410 @@
+            registry.run(world, system);
+        });
+    }
+
+    /// Run the systems with the provided [`SystemId`].
+    ///
+    /// Calls [`SystemRegistry::run_by_id`].
+    #[inline]
+    pub fn run_system_by_id(&mut self, id: SystemId) -> Result<(), SystemRegistryError> {
+        if !self.contains_resource::<SystemRegistry>() {
+            panic!(
+                "SystemRegistry not found: Nested and recursive one-shot systems are not supported"
+            );
+        }
+
+        self.resource_scope(|world, mut registry: Mut<SystemRegistry>| {
+            registry.run_by_id(world, id)
+        })
+    }
+}
+
+/// The [`Command`] type for [`SystemRegistry::run_system`]
+#[derive(Debug, Clone)]
+pub struct RunSystem<M: Send + Sync + 'static, S: IntoSystem<(), (), M> + Send + Sync + 'static> {
+    _phantom_marker: PhantomData<M>,
+    system: S,
+}
+
+impl<M: Send + Sync + 'static, S: IntoSystem<(), (), M> + Send + Sync + 'static> RunSystem<M, S> {
+    /// Creates a new [`Command`] struct, which can be added to [`Commands`](crate::system::Commands)
+    #[inline]
+    #[must_use]
+    pub fn new(system: S) -> Self {
+        Self {
+            _phantom_marker: PhantomData::default(),
+            system,
+        }
+    }
+}
+
+impl<M: Send + Sync + 'static, S: IntoSystem<(), (), M> + Send + Sync + 'static> Command
+    for RunSystem<M, S>
+{
+    #[inline]
+    fn apply(self, world: &mut World) {
+        world.run_system(self.system);
+    }
+}
+
+/// The [`Command`] type for [`SystemRegistry::run_by_id`].
+#[derive(Debug, Clone)]
+pub struct RunSystemById {
+    system_id: SystemId,
+}
+
+impl RunSystemById {
+    /// Creates a new [`Command`] struct, which can be added to [`Commands`](crate::system::Commands)
+    pub fn new(system_id: SystemId) -> Self {
+        Self { system_id }
+    }
+}
+
+impl Command for RunSystemById {
+    #[inline]
+    fn apply(self, world: &mut World) {
+        if !world.contains_resource::<SystemRegistry>() {
+            panic!(
+                "SystemRegistry not found: Nested and recursive one-shot systems are not supported"
+            );
+        }
+
+        world.resource_scope(|world, mut registry: Mut<SystemRegistry>| {
+            registry
+                .run_by_id(world, self.system_id)
+                // Ideally this error should be handled more gracefully,
+                // but that's blocked on a full error handling solution for commands
+                .unwrap();
+        });
+    }
+}
+
+/// An operation on a [`SystemRegistry`] failed
+#[derive(Debug)]
+pub enum SystemRegistryError {
+    /// A system was run by label, but no system with that label was found.
+    ///
+    /// Did you forget to register it?
+    SystemIdNotRegistered(SystemId),
+}
+
+mod tests {
+    use crate as bevy_ecs;
+    use crate::prelude::*;
+
+    #[derive(Resource, Default, PartialEq, Debug)]
+    struct Counter(u8);
+
+    #[allow(dead_code)]
```

Comment:
This needs a comment about why it's required.

Reply:
It wasn't required, I've removed it